src/test/clojure/clara/test_rules.clj

```diff
@@ -5025,3 +5029,31 @@
     (is (= (query end-session-equal-facts lousy-weather-query)
            (query end-session-identical-facts lousy-weather-query)
            [{}]))))
+
+(deftest test-alpha-batching-with-multiple-ancestors
+  ;; see issue 257 for details of the bug surronding this test
+  (let [accum-state (atom #{})
+        
+        stateful-accum (acc/accum
+                         {:initial-value []
+                          :reduce-fn conj
+                          :retract-fn (fn [items retracted] (remove #{retracted} items))
+                          :convert-return-fn (fn [items] 
+                                               (do 
+                                                 (swap! accum-state conj items)
+                                                 items))})
+        
+        map-rule (dsl/parse-rule [[?maps <- stateful-accum :from [List]]]
+                                 ;; don't care about whats inserted
+                                 (->LousyWeather))
+        
+        identity-list (ArrayList.)
+        linked-list (LinkedList.)
+        
+        ses (-> (mk-session [map-rule])
+              (insert-all [identity-list linked-list])
+              (fire-rules))]
+    
```

Comment:
The issue should work for more a more detailed explanation - +1 to the comment referencing it - but could you add a brief comment here about why you're asserting this?  Something like "Validate that two distinct fact types are batched together when passed to a condition on a common ancestor type.  We use an accumulator because it provides a convenient way to directly access the facts passed to the condition without exposing Clara internals."